### PR TITLE
Ports: Add port aria2 version 1.36.0

### PR DESCRIPTION
**Pull request by EWouters**
I ported aria2, however I can't get it to work.

- [x] ~~Fails to find `libxml2` so I have to pass `--without-libxml2`~~ (fixed in https://github.com/SerenityOS/serenity/commit/139f9039607ef3972a11b2085ce798f9759aac9d)
- [ ] Error `--max-mmap-limit`
image

### CODE REVIEW

**Comment by EWouters** [comment]
Currently blocked on #13915.

**Comment by timschumi** [comment]
It might be worth checking if `libxml2` can now be found after 139f9039607ef3972a11b2085ce798f9759aac9d has been merged.

**Comment by EWouters** [comment]
It can find `libxml2` fine now,  but the numeric over/under-flow error is still there.

File: Ports/aria2/package.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port="aria2"
+version="1.36.0"
```
Comment (timschumi) [inline Ports/aria2/package.sh:3]: Should be single quotes, same with `auth_type` and `useconfigure`.

File: Ports/aria2/package.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+files="https://github.com/aria2/aria2/releases/download/release-${version}/aria2-${version}.tar.xz ${port}-${version}.tar.xz 58d1e7608c12404f0229a3d9a4953d0d00c18040504498b483305bcb3de907a5"
+auth_type="sha256"
+depends=("libssh2" "libxml2" "openssl" "zlib" "libuv")
+useconfigure="true"
+configopts+=("--with-libuv")
```
Comment (timschumi) [inline Ports/aria2/package.sh:8]: Needs `--with-sysroot=${SERENITY_INSTALL_ROOT}` and should be split up across multiple lines.

File: Ports/aria2/package.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+pre_configure() {
+    run autoreconf -i
+}
```
Comment (timschumi) [inline Ports/aria2/package.sh:12]: This is not needed, the tarball ships in an already configured state.

File: Ports/aria2/patches/0001-Add-Serenity-to-config.sub.patch [hunk 4]
```diff
@@ -0,0 +1,37 @@
+From 9308979ce6065bcb83cbd76f3d7eec4bc672973a Mon Sep 17 00:00:00 2001
+From: EWouters <6179932+EWouters@users.noreply.github.com>
+Date: Thu, 14 Apr 2022 00:19:04 +0200
+Subject: [PATCH 1/2] Add Serenity to config.sub
```
Comment (timschumi) [inline Ports/aria2/patches/0001-Add-Serenity-to-config.sub.patch:4]: Please replace this patch with `config_sub_paths` in the `package.sh` script.

File: Ports/aria2/patches/0002-Include-sys-select.h.patch [hunk 5]
```diff
@@ -0,0 +1,24 @@
+From 839f9ff4a68a40d77e873a63531394eb4af8e50b Mon Sep 17 00:00:00 2001
```
Comment (timschumi) [inline Ports/aria2/patches/0002-Include-sys-select.h.patch:1]: Please regenerate this using a current version of `./package.sh dev` to get rid of the unnecessary metadata.

**Review by timschumi — CHANGES_REQUESTED** [review]
The number parsing issue should be fixed by #14313.

**Comment by EWouters** [comment]
It somewhat works now :^)

image

**Pull request closed without merging**